pkg/kubelet/status/status_manager.go

```diff
@@ -774,7 +777,12 @@ func (m *manager) syncPod(uid types.UID, status versionedPodStatus) {
 		return
 	}
 
-	mergedStatus := mergePodStatus(pod.Status, status.status, m.podDeletionSafety.PodCouldHaveRunningContainers(pod))
+	podMightNeedToUnprepareResources := false
+	if utilfeature.DefaultFeatureGate.Enabled(features.DynamicResourceAllocation) {
+		podMightNeedToUnprepareResources = m.podDeletionSafety.PodMightNeedToUnprepareResources(pod.UID)
+	}
+
+	mergedStatus := mergePodStatus(pod.Status, status.status, m.podDeletionSafety.PodCouldHaveRunningContainers(pod), podMightNeedToUnprepareResources)
```

Comment:
Based on previous comments this can be changed to:
```suggestion
	delayTransitioningToTerminal := m.podDeletionSafety.PodCouldHaveRunningContainers(pod)
	if utilfeature.DefaultFeatureGate.Enabled(features.DynamicResourceAllocation) {
		delayTransitioningToTerminal = delayTransitioningToTerminal || m.podDeletionSafety.PodMightNeedToUnprepareResources(pod.UID)
	}

	mergedStatus := mergePodStatus(pod.Status, status.status, delayTransitioningToTerminal)
```